cfpb/design-stories
# Set up build and packaging tooling in preparation for publishing

Right now we're using storybook (`yarn start` which runs `yarn storybook`) to build and test our components. This works great as a self-contained workshop but there's no way for an external project to import and use the components. Running `yarn build` throws an error.

This PR adds an entrypoint file and updates our vite build config to compile everything into the `dist/` dir. Once things are stabilized we'll start versioning and publishing the package to npm.

## Changes

- Update docs
- Adds an index file that exports everything
- Clean up vite config file

## How to test this PR

1. Pull down this branch.
1. `yarn && yarn build`
1. There should be a bunch of files in `dist/`. I think there are lots of extraneous files there but we can tweak the build config down the road.

You can also test it with a real app. The sbl-prototype repo is [configured](https://github.com/cfpb/sbl-prototype/blob/main/app/package.json#L35) to [import](https://github.com/cfpb/sbl-prototype/blob/main/app/src/pages/Home.tsx#L6) this branch. Try running it with:

```
git clone https://github.com/cfpb/sbl-prototype.git
cd sbl-prototype/app
yarn
yarn dev
```

And it should open http://localhost:8881/ with a green notification and some buttons that are imported from this repo. Note: that repo uses yarn in [pnp mode](https://yarnpkg.com/features/pnp) mode so you'll need yarn v3. If you don't want to install that, try the [docker steps](https://github.com/cfpb/sbl-prototype#how-to-use).
